src/common/args.cpp

```diff
@@ -288,8 +296,8 @@ fs::path ArgsManager::GetBlocksDirPath() const
     // this function
     if (!path.empty()) return path;
 
-    if (IsArgSet("-blocksdir")) {
-        path = fs::absolute(GetPathArg("-blocksdir"));
+    if (!BlocksDirSetting::Value(*this).isNull()) {
+        path = fs::absolute(BlocksDirSettingPath::Get(*this));
```

Comment:
Not sure what to make of `BlocksDirSetting`/`BlocksDirSettingPath` pairs..
- `BlocksdirSettingPath` is not registered to avoid collisions.
- I guess you are avoidant when it comes to adding types.. but this seems worse.

Would suggest:
```suggestion
    if (!BlocksDirSetting::Value(*this).isNull()) {
        path = fs::absolute(BlocksDirSetting::Get(*this));
```
and having `BlocksDirSetting` contain an `fs::path` directly.